pkg/fileReader/reader.go

```diff
@@ -55,16 +56,22 @@ func CreateFileReader(opts *FileReaderOptions) *FileReader {
 	return fileReader
 }
 
-func (fr *FileReader) FilterFiles(paths []string) ([]string, error) {
+func (fr *FileReader) FilterFiles(paths []string, excludePattern string) ([]string, error) {
 	var filePaths []string
 
+	excludeRe, err := regexp.Compile(excludePattern)
```

Comment:
just a suggestion :)
```suggestion
	excludeRegex, err := regexp.Compile(excludePattern)
```